src/app/components/overlay/overlay.ts

```diff
@@ -0,0 +1,249 @@
+    }
+
+    @Input() set overlayDirection(value: string) {
+        const viewport = DomHandler.getViewport();
+        if (value && viewport.width < this.overlayBreakpoints) {
+            this._overlayDirection = value;
+            switch (value) {
+                case 'start':
+                    this.transformOptions = "translate3d(0px, -100%, 0px)";
+                break;
+                case 'end':
+                    this.transformOptions = "translate3d(0px, 100%, 0px)";
+                break;
+                case 'center': 
+                    this.transformOptions = "scale(0.8)";
+                break;
+            }
+        } 
+        else {
+            this.transformOptions = "scaleY(0.8)";
+        }
+    }
+
+    @Input() set appendTo(val) {
+        this._appendTo = val;
+    }
+
+    get appendTo(): any {
+        return this._appendTo;
+    }
+
+    get overlayBreakpoints() {
+        return this.config.overlayOptions ? this.config.overlayOptions.breakpoint : null;
+    }
+
+    @ViewChild('overlay') overlayViewChild: ElementRef;
+
+    _overlayDirection: string;
+
+    _overlayBreakpoints: number;
+
+    _appendTo: any;
+
+    transformOptions: string = "scaleY(0.8)";
+
+    mask: HTMLDivElement;
+
+    maskClickListener: Function;
+
+    constructor(@Inject(DOCUMENT) private document: Document, public el: ElementRef, public renderer: Renderer2, private config: PrimeNGConfig, public overlayService: OverlayService) {}
+
+    onOverlayClick(event: MouseEvent) {
+        this.overlayService.add({
+            originalEvent: event,
+            target: this.el.nativeElement
+        });
+    }
+
+    appendOverlay() {
+        const viewport = DomHandler.getViewport();
+        if (this.autoZIndex) {
+            ZIndexUtils.set('overlay', this.overlayViewChild.nativeElement, this.baseZIndex + this.config.zIndex.overlay);
+        }
+        if (viewport.width < this.overlayBreakpoints) {
+            this.appendMask();
+            DomHandler.addClass(this.document.body, 'p-overflow-hidden');
+            DomHandler.addClass(this.overlayViewChild.nativeElement, 'p-overlay-responsive');
+            this.mask.appendChild(this.el.nativeElement);
+        }
+        else {
+            if (this.appendTo) {
+                if (this.appendTo === 'body') {
+                    this.document.body.appendChild(this.el.nativeElement);
+                }
+                else {
+                    DomHandler.appendChild(this.el.nativeElement, this.appendTo);
+                }
+            }
+        }
+
+        if (!this.overlayViewChild.nativeElement.style.minWidth) {
+            this.overlayViewChild.nativeElement.style.minWidth = DomHandler.getWidth(this.container) + 'px';
+        }
+    }
+
+    alignOverlay() {
+        const viewport = DomHandler.getViewport();
+
+        if(this.overlayViewChild) {
+            if (viewport.width < this.overlayBreakpoints) {
+                switch (this.overlayDirection) {
```

Comment:
also mb create method

```ts
DomHandler.addClass(this.el.nativeElement, this.getOverlayDirection(this.overlayDirection))
```

Reply:
Hi @volvachev 

Thanks for your feedback, it's not the production version of the code it'll be updated and refactored before release.